src/test/kotlin/note/application/NoteUpdaterTest.kt

```diff
@@ -0,0 +1,65 @@
+package note.application
+
+import note.domain.*
+import note.domain.exceptions.NonExistentNoteException
+import note.mothers.IdentifierMother
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.mockito.Mockito
+
+class ReminderUpdaterTest {
+    private lateinit var reminderUpdater: ReminderUpdater
+    private lateinit var generator: NoteIdentifierGenerator
+    private lateinit var repository: NoteRepository
+
+    private val newTitle = "title"
+    private val newDescription = "description"
+    private val identifier = IdentifierMother.getValidIdentifier()
+
+    @BeforeEach
+    fun setUp() {
+        repository = Mockito.mock(NoteRepository::class.java)
+        generator = Mockito.mock(NoteIdentifierGenerator::class.java)
+        reminderUpdater = ReminderUpdater(repository, generator)
+    }
+
+    @Test
+    fun `Invalid identifier throws NonExistentReminderException and doesn't call save`() {
+        // Initialize
+        val identifier = IdentifierMother.getValidIdentifier()
+        Mockito.`when`(repository.search(identifier)).thenReturn(null)
+
+        // Execute and Assert
+        assertThrows<NonExistentNoteException> {
+            reminderUpdater.update(identifier, newTitle, newDescription)
+        }
+
+        // Assert number of calls
+        assertNumberOfCalls(1, 0, 0)
```

Comment:
We shouldn't watch all the number of calls on the repository, only the ones that matter.

If the user puts an invalid identifier, then we should check that doesn't save anything on the repository. We don't check if it searches it on the repository. We only:
- Mock the repository to return null if they ask us about the given identifier.
- Assert that nothing is saved.

All the other assertions are implementation details.